[no-test] Accessible navtabs in services

marusak opened the pull request:
This PR tries to make navtabs in services page accessible.
The accessibility is implemented two ways - (in both cases enter and space select current tab)

1. As bunch on toggle buttons, which is simpler but does not make much sense as it may seem that more of them can be selected at once
2. As navtabs. It is possible to move with left and right arrow. And makes more sense.

Then there is implementation of focus. It is implemented that firstly is focused selection of type of unit and when selected search field is focused. This makes sense for users with screenreaders (because when search field is focused first, then it would be very hard to discover that there is something before that) but for other users it may seem as there is nothing focused (but if it would be discoverable that hitting space or enter would jump to search box...)

Also last commit moves default selected tab as first. This is the same logic about discovering elements on left while using screen readers.

@garrett definitely will have options about this and will explain to me if this is the right direction or not :)

garrett commented on pkg/systemd/services.css:
```diff
@@ -118,3 +118,7 @@ table.systemd-unit-relationship-table td:first-child {
 }
+
+.nav-tabs li {
+  outline: none;
```
We _definitely_ want an outline of the focused element somewhere, BTW. So if we remove it here, it needs to be done elsewhere...

See:
- https://a11yproject.com/posts/never-remove-css-outlines/
- https://css-tricks.com/removing-the-dotted-outline/

garrett commented:
Quoting myself from https://github.com/cockpit-project/cockpit/pull/11241#pullrequestreview-210038183:

> Ah, cool to see tabindex as the tabs need to be selected — but why is it in `<li>` instead of `<a>`? (Better yet would be to add `href` to the `<a>`, instead of using a `tabindex`, and use [a URL fragment*](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Attributes) to remember the state in the URL.)
>
> _* URL fragment (`#this-thing-here`), not to be confused with a Document Fragment, React.Fragment, CSS Fragmentation, or anything else called "fragment" in webdev._ :wink:

garrett commented:
Oh, by-the-way, yes `:focus-within` targets a parent element based on the state of children elements. It's the first CSS selector to do so. (We don't have [`:has()`](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) implemented anywhere yet, but this is effectively like `:has(:focus)`.)

marusak commented:
@garrett thank you for the changes you pushed. I picked them up and merged into other commits. Also I squashed all commits that made sense to be in one. Now it is only 3 commits. Also I've rebased to master as there were conflicts.
Can you please review?

garrett commented:
@marusak: Thanks! I did have some conflicts and tried to sort them out... but I might've messed things up. I hope you were able to fix any of the issues or regressions I might've introduced.

Checking now.

garrett commented:
(Oh, I just realized I pushed broken code but forgot to push a local change I did after. Sorry about that. Thanks again for fixing it!)

garrett commented:
I guess we need to rebase on #11334?

marusak commented:
> I guess we need to rebase on #11334?

Yes, once merged I will rebase and fix the issue mentioned [here](https://github.com/cockpit-project/cockpit/pull/11334#pullrequestreview-227692048)

marusak commented:
Blocked by #11334

martinpitt commented:
#11334 landed, so unblocked. But has a conflict, so needs a rebase now.

marusak commented:
I've rebased it. Can you please review @garrett ?

marusak commented:
I've implemented this by using PF `Tabs` and therefore there is no need to do the accessibility manually.
Obsoleted by #12349
